cluster-provision/gocli/docker/docker.go

```diff
@@ -232,6 +234,15 @@ func NewCleanupHandler(cli *client.Client, cleanupChan chan error, errWriter io.
 						err := cli.ContainerRemove(ctx, c, types.ContainerRemoveOptions{Force: true})
 						if err != nil {
 							fmt.Fprintf(errWriter, "%v\n", err)
+							failedContainers = append(failedContainers, c)
+						}
+					}
+					if failedContainers != nil {
```

Comment:
no need the condition, as the range would just run on empty list, isn't it ?

Maybe worth renaming to something like containerRemoveRetry / containerRemoveFailed ?